### PR TITLE
[ASCollectionView] Call -invalidateFlowLayoutDelegateMetrics when rotating. #trivial

### PR DESCRIPTION
**Pull request by appleguy**
This allows rotation to work properly when rotating UIKit passthrough
cells that need to change width.

This diff was written and tested for use with #431, but it's fundamentally the same with the current HEAD.

### CODE REVIEW

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Source/ASCollectionView.mm [hunk 1]
```diff
@@ -2246,32 +2247,32 @@ - (void)didMoveToWindow
...
   // appearance update, we do not need to relayout all nodes.
   // For a more permanent fix to the unsafety mentioned above, see https://github.com/facebook/AsyncDisplayKit/pull/2182
   ASScrollDirection scrollDirection = self.scrollableDirections;
-  BOOL fixedVertically = (ASScrollDirectionContainsVerticalDirection(scrollDirection) == NO);
+  BOOL fixedVertically   = (ASScrollDirectionContainsVerticalDirection  (scrollDirection) == NO);
   BOOL fixedHorizontally = (ASScrollDirectionContainsHorizontalDirection(scrollDirection) == NO);
 
-  BOOL changedInNonScrollingDirection = (fixedHorizontally && newBounds.size.width != lastUsedSize.width) || (fixedVertically && newBounds.size.height != lastUsedSize.height);
+  BOOL changedInNonScrollingDirection = (fixedHorizontally && newSize.width  != lastUsedSize.width) ||
+                                        (fixedVertically   && newSize.height != lastUsedSize.height);
```
Comment (Adlai-Holler) [inline Source/ASCollectionView.mm]: Delightful!

File: Source/ASCollectionView.mm [hunk 2]
```diff
@@ -367,6 +367,7 @@ - (void)scrollToItemAtIndexPath:(NSIndexPath *)indexPath atScrollPosition:(UICol
 
 - (void)relayoutItems
 {
+  [self invalidateFlowLayoutDelegateMetrics];
```
Comment (nguyenhuy) [inline Source/ASCollectionView.mm]: @appleguy Now that #431 is merged, please update this PR. Also, should `[self invalidateFlowLayoutDelegateMetrics];` be called inside the invalidation block?

File: Source/ASCollectionView.mm (hunk 1 above)
Comment (appleguy) [reply Source/ASCollectionView.mm]: :-D thanks for sharing your aesthetic approval, always good to see & re-align as a team

**Comment by appleguy** [comment]
@nguyenhuy @Adlai-Holler thanks guys - ready for another look

**Pull request merged**